src/main/java/org/tugraz/sysds/runtime/controlprogram/context/ExecutionContext.java

```
@@ -651,4 +654,37 @@ public void cleanupCacheableData(CacheableData<?> mo) {
 			throw new DMLRuntimeException(ex);
 		}
 	}
+	
+	public void initLastBranch(){
+		_lastBranch = new BitSet();
+		_branchCounter = 0;
+	}
+	
+	public void removeLastBranch(){
+		_lastBranch = null;
+		_branchCounter = 0;
+	}
+	
+	public void clearLastBranch(){
+		if (_lastBranch != null)
+			_lastBranch.clear();
+		_branchCounter = 0;
+	}
+	
+	public void setBranchPredicateValue(Boolean value){
+		if (_lastBranch != null){
+			if (value)
+				_lastBranch.set(_branchCounter);
+			_branchCounter++;
+		}
+	}
+	
+	public Long getLastBranch(){
+		if (_lastBranch != null)
+			if (_lastBranch.toLongArray().length == 1)
+				return _lastBranch.toLongArray()[0];
```

Please raise an exception if the length exceeds 1